File: csharp/P0217_ContainsDuplicate/Solutions/HashSetLookup.cs
namespace P0217_ContainsDuplicate.Solutions.HashSetLookup;

public class Solution
{
    public bool ContainsDuplicate(int[] nums)
    {
        var seen = new HashSet<int>();

        foreach (var num in nums)
            if (!seen.Add(num))  //.Add() returns false if the item exists
                return true;
        
        return false;
    }
}

File: rust/p0217_contains_duplicate/src/hashset_lookup.rs
use std::collections::HashSet;

pub struct Solution;

impl Solution {
    pub fn contains_duplicate(nums: Vec<i32>) -> bool {
        let mut seen: HashSet<i32> = HashSet::new();

        for num in nums {
            if seen.contains(&num) {
                return true;
            }
            seen.insert(num);
        }
        false
    }
}
